test/e2e/helpers_gateway_test.go

```diff
@@ -0,0 +1,303 @@
+	_, err = env.Cluster().Client().CoreV1().Services(corev1.NamespaceDefault).Create(ctx, service, metav1.CreateOptions{})
+	require.NoError(t, err)
+
+	t.Logf("creating an HTTPRoute for service %s with Gateway %s", service.Name, gw.Name)
+	pathMatchPrefix := gatewayv1alpha2.PathMatchPathPrefix
+	path := "/httpbin"
+	httpPort := gatewayv1alpha2.PortNumber(80)
+	httproute := &gatewayv1alpha2.HTTPRoute{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: uuid.NewString(),
+			Annotations: map[string]string{
+				annotations.AnnotationPrefix + annotations.StripPathKey: "true",
+			},
+		},
+		Spec: gatewayv1alpha2.HTTPRouteSpec{
+			CommonRouteSpec: gatewayv1alpha2.CommonRouteSpec{
+				ParentRefs: []gatewayv1alpha2.ParentReference{{
+					Name: gatewayv1alpha2.ObjectName(gw.Name),
+				}},
+			},
+			Rules: []gatewayv1alpha2.HTTPRouteRule{{
+				Matches: []gatewayv1alpha2.HTTPRouteMatch{{
+					Path: &gatewayv1alpha2.HTTPPathMatch{
+						Type:  &pathMatchPrefix,
+						Value: &path,
+					},
+				}},
+				BackendRefs: []gatewayv1alpha2.HTTPBackendRef{{
+					BackendRef: gatewayv1alpha2.BackendRef{
+						BackendObjectReference: gatewayv1alpha2.BackendObjectReference{
+							Name: gatewayv1alpha2.ObjectName(service.Name),
+							Port: &httpPort,
+						},
+					},
+				}},
+			}},
+		},
+	}
+	_, err = gc.GatewayV1alpha2().HTTPRoutes(corev1.NamespaceDefault).Create(ctx, httproute, metav1.CreateOptions{})
+	require.NoError(t, err)
+}
+
+// verifyHTTPRoute verifies an HTTPRoute exposes a route at /httpbin
+// TODO this is not actually specific to HTTPRoutes. It is verifyIngress with the KongIngress removed
+// Once we support HTTPMethod HTTPRouteMatch handling, we can combine the two into a single generic function.
+func verifyHTTPRoute(ctx context.Context, t *testing.T, env environments.Environment) {
+	t.Log("finding the kong proxy service ip")
+	svc, err := env.Cluster().Client().CoreV1().Services(namespace).Get(ctx, "kong-proxy", metav1.GetOptions{})
+	require.NoError(t, err)
+	proxyIP := getKongProxyIP(ctx, t, env, svc)
+
+	t.Logf("waiting for route from Ingress to be operational at http://%s/httpbin", proxyIP)
+	httpc := http.Client{Timeout: time.Second * 10}
+	require.Eventually(t, func() bool {
+		resp, err := httpc.Get(fmt.Sprintf("http://%s/httpbin", proxyIP))
+		if err != nil {
+			return false
+		}
+		defer resp.Body.Close()
+		if resp.StatusCode == http.StatusOK {
+			b := new(bytes.Buffer)
+			n, err := b.ReadFrom(resp.Body)
+			require.NoError(t, err)
+			require.True(t, n > 0)
+			return strings.Contains(b.String(), "<title>httpbin.org</title>")
+		}
+		return false
+	}, ingressWait, time.Second)
+}
+
+// deployTCPRoute creates a `TCPRoute` and related backend deployement/service.
+func deployTCPRoute(ctx context.Context, t *testing.T, env environments.Environment, gw *gatewayv1alpha2.Gateway) {
+	gc, err := gatewayclient.NewForConfig(env.Cluster().Config())
+	assert.NoError(t, err)
```

Comment:
`assert`? So if there's an error we continue? I don't believe there would be a point in that. What did you have in mind here?